File: src/Entity/Demandedons.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;




#[ORM\Table(name: "demandedons", indexes: [
    new ORM\Index(name: "idUtilisateur", columns: ["idUtilisateur"]),
    new ORM\Index(name: "idDons", columns: ["idDons"])

])]
#[ORM\Entity]
class Demandedons
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    #[ORM\Column(name: "idDemande", type: "integer", nullable: false)]
    private ?int $idDemande;

    #[ORM\Column(name: "contenu", type: "text", length: 65535, nullable: true, options: ["default" => "NULL"])]
   

    private ?string $contenu = null;

    #[ORM\Column(name: "image", type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private ?string $image = null;

    #[ORM\Column(name: "datePublication", type: "datetime", nullable: false)]

    private \DateTimeInterface $datePublication;

    #[ORM\Column(name: "nbpoints", type: "integer", nullable: true)]
    private ?int $nbpoints = null;

    #[ORM\Column(name: "nomuser", type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private ?string $nomuser = null;

    #[ORM\Column(name: "prenomuser", type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private ?string $prenomuser = null;

    #[ORM\Column(name: "objectifPoints", type: "integer", nullable: true)]
 
    private ?int $objectifPoints;
    

    #[ORM\Column(name: "delai", type: "date", nullable: true)]

    private ?\DateTimeInterface $delai;

    #[ORM\ManyToOne(targetEntity: "Dons")]
    #[ORM\JoinColumn(name: "idDons", referencedColumnName: "idDons")]
    private ?Dons $idDons;

    #[ORM\ManyToOne(targetEntity: "Utilisateur")]
    #[ORM\JoinColumn(name: "idUtilisateur", referencedColumnName: "idUser")]
    private ?Utilisateur $idUtilisateur;

    public function getIdDemande(): ?int
    {
        return $this->idDemande;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(?string $contenu): self
    {
        $this->contenu = $contenu;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getDatePublication(): ?\DateTimeInterface
    {
        return $this->datePublication;
    }

    public function setDatePublication(\DateTimeInterface $datePublication): self
    {
        $this->datePublication = $datePublication;
        return $this;
    }

    public function getNbpoints(): ?int
    {
        return $this->nbpoints;
    }

    public function setNbpoints(?int $nbpoints): self
    {
        $this->nbpoints = $nbpoints;
        return $this;
    }

    public function getNomuser(): ?string
    {
        return $this->nomuser;
    }

    public function setNomuser(?string $nomuser): self
    {
        $this->nomuser = $nomuser;
        return $this;
    }

    public function getPrenomuser(): ?string
    {
        return $this->prenomuser;
    }

    public function setPrenomuser(?string $prenomuser): self
    {
        $this->prenomuser = $prenomuser;
        return $this;
    }

    public function getIdDons(): ?Dons
    {
        return $this->idDons;
    }

    public function setIdDons(?Dons $idDons): self
    {
        $this->idDons = $idDons;
        return $this;
    }

    public function getIdUtilisateur(): ?Utilisateur
    {
        return $this->idUtilisateur;
    }

    public function setIdUtilisateur(?Utilisateur $idUtilisateur): self
    {
        $this->idUtilisateur = $idUtilisateur;
        return $this;
    }

    public function getObjectifPoints(): ?int
    {
        return $this->objectifPoints;
    }

    public function setObjectifPoints(?int $objectifPoints): self
    {
        $this->objectifPoints = $objectifPoints;
        return $this;
    }

    public function getDelai(): ?\DateTimeInterface
    {
        return $this->delai;
    }

    public function setDelai(?\DateTimeInterface $delai): self
    {
        $this->delai = $delai;
        return $this;
    }
}

File: src/main/java/entities/DemandeDons.java
package entities;

import javafx.beans.property.IntegerProperty;
import javafx.beans.property.SimpleIntegerProperty;

import java.sql.Timestamp;

public class DemandeDons {
    private final IntegerProperty idDemandeProperty = new SimpleIntegerProperty();

    private int idDemande;
    private int idDons;
    private int idUtilisateur;
    private String contenu;
    private String image;
    private Timestamp datePublication;
    private String nomUser;
    private String prenomUser;
    private int nbPoints;

    private int idUser;

    public int getTotalPointsGagnes() {
        return totalPointsGagnes;
    }

    public void setTotalPointsGagnes(int totalPointsGagnes) {
        this.totalPointsGagnes = totalPointsGagnes;
    }

    private int totalPointsGagnes;




    public DemandeDons(int idDemande, int idUtilisateur, String contenu, String image, Timestamp datePublication) {
        this.idDemande = idDemande;
        this.idUtilisateur = idUtilisateur;
        this.contenu = contenu;
        this.image = image;
        this.datePublication = datePublication;
    }

    public DemandeDons() {

    }
    public IntegerProperty idDemandeProperty() {
        return idDemandeProperty;
    }
    // Getters and Setters
    public int getIdDemande() {
        return idDemande;
    }

    public void setIdDemande(int idDemande) {
        this.idDemande = idDemande;
    }

    public int getIdUtilisateur() {
        return idUtilisateur;
    }

    public void setIdUtilisateur(int idUtilisateur) {
        this.idUtilisateur = idUtilisateur;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Timestamp getDatePublication() {
        return datePublication;
    }

    public void setDatePublication(Timestamp datePublication) {
        this.datePublication = datePublication;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public String getNomUser() {
        return nomUser;
    }

    public void setNomUser(String nomUser) {
        this.nomUser = nomUser;
    }

    public String getPrenomUser() {
        return prenomUser;
    }

    public void setPrenomUser(String prenomUser) {
        this.prenomUser = prenomUser;
    }

    public int getNbPoints() {
        return nbPoints;
    }

    public void setNbPoints(int nbPoints) {
        this.nbPoints = nbPoints;
    }



    public int getIdDons() {
        return idDons;
    }
}
